### DISCUSSION TITLE
Is there a way to tell it the number of characters the password has?

### DISCUSSION THREAD

**Discussion by FunkyJustin** [illsk1lls/ZipRipper#26] [Q&A]
I know the number of characters the password has in the file I'm trying to open but I don't see any obvious way to letting the program know so that it can potentially shorten the brute-forcing time. Is this already implemented somewhere?

**Comment by illsk1lls** [comment]
Rule options will be the next feature added

**Reply by flicosh** [reply]
Has this feature been added yet, I too would like to specify the number of characters in order to speed up the Brute Force attack.

**Reply by illsk1lls** [reply]
what is the filetype? and do you have a GPU? (theres isnt a way yet but I'll tell you what line to edit for your attempt)

**Reply by flicosh** [reply]
It's a Zip file, I have a NVIDIA GeForce RTX 4050 Laptop GPU which seem to be trying about 44000 passwords per second.

Unfortunately the password is 16 characters, so the chances of cracking it are vanishingly small, but you never know!

**Reply by illsk1lls** [reply]
For your filetype and card try changing line #620 from:
`					SET "FLAG=--format=ZIP-opencl"`
to:
`					SET "FLAG=--format=ZIP-opencl --min-length=16 --max-length=16"`

That will limit the pw length attempts to 16, although it wont alert you, you will be able to tell it is working correctly by pressing any key for status, and looking at the length of the guesses

**Reply by flicosh** [reply]
That works just great thanks :-)

Now I'm getting the output reports I'd expect while it's searching:
0g 0:00:19:42  0g/s 44325p/s 2759Kc/s 2759KC/s ie111aeeeeaa11aa..ie11e11e111eaea1

**Reply by illsk1lls** [reply]
no problem.  i'm still trying to figure out a clean way to work it in when time permits

**Comment by juanolvera7735** [comment]
How about it one does not have a GPU?  I have a .7z file with a 10-character password.

**Comment by illsk1lls** [comment]
Change line #337 from

`john --wordlist="%ProgramData%\JtR\run\password.lst" --rules=single,all "%ProgramData%\JtR\run\pwhash" !FLAG!`
to
`john --wordlist="%ProgramData%\JtR\run\password.lst" --rules=single,all --min-length=10 --max-length=10 "%ProgramData%\JtR\run\pwhash" !FLAG!`

and if you want to do bruteforce and use an empty TXT file as a wordlist, change line #335 from
`john "%ProgramData%\JtR\run\pwhash" --incremental=ASCII.chr !FLAG!`
to
`john "%ProgramData%\JtR\run\pwhash" --incremental=ASCII.chr --min-length=10 --max-length=10 !FLAG!`

*Keep in mind as I do updates to the script the positioning of these lines may change and this comment will become outdated*

**Comment by RAPIRA01** [comment]
У меня rar файл с 12 значным паролем, я все сделал как указанно выше но не помогло( 
![20251016_214602](https://github.com/user-attachments/assets/46ae3d45-1f14-4733-91ee-cf66a3de0864)
